library/core/src/iter/adapters/array_chunks.rs

```diff
@@ -0,0 +1,74 @@
+use crate::iter::{DoubleEndedIterator, Fuse, FusedIterator, Iterator, TrustedLen};
+
+use crate::array;
+
+/// Returns an iterator over `N` elements of the iterator at a time.
+///
+/// The chunks do not overlap. If `N` does not divide the length of the
+/// iterator, then the last up to `N-1` elements will be omitted.
+#[derive(Debug, Clone)]
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+#[unstable(feature = "iter_array_chunks", reason = "recently added", issue = "none")]
+pub struct ArrayChunks<I, const N: usize> {
+    iter: Fuse<I>,
+}
+
+impl<I, const N: usize> ArrayChunks<I, N>
+where
+    I: Iterator,
+{
+    #[inline]
+    pub(in crate::iter) fn new(iter: I) -> Self {
+        assert!(N != 0);
```

Comment:
`array_chunks()` should document that `N==0` will panic. We may also want an explicit panic that states the cause.